pkg/sql/sqlstats/insights/ingester.go

```diff
@@ -31,12 +31,26 @@ type concurrentBufferIngester struct {
 		eventBuffer *eventBuffer
 	}
 
+	testKnobs struct {
+		// noFlush prevents signals on eventBufferCh, which prevents flushing of
+		// the eventBuffer, unless Clear() is called.
+		noFlush bool
+	}
+
 	eventBufferCh chan *eventBuffer
 	registry      *lockingRegistry
 	running       uint64
 }
 
-var _ Writer = &concurrentBufferIngester{}
+// Clear flushes the underlying buffer, and signals the underlying registry
+// to clear any remaining cached data afterwards. This is a synchronous operation.
+func (i *concurrentBufferIngester) Clear(ctx context.Context) {
+	i.ingest(ctx, i.guard.eventBuffer)
```

Comment:
Eh, this is wrong the more I think about it. This should be done within the context of the async task that listens on the eventBufferCh. I'll put up a patch.

Reply:
Done.